_P208_Nokia_LCD_5110.ino

```diff
@@ -0,0 +1,303 @@
+//#######################################################################################################
+//################################## Plugin 208: NOKIA 5110 lcd #########################################
+//#######################################################################################################
+#include "_Plugin_Helper.h"
+#ifdef USES_P208
+
+// Working:
+// - support different digit-size's 
+// - Display Text via:
+//   1.  ESPEasy-Webinterface (Line-1-Line-x)
+//   2.  http-request:
+//      - BackLight on  via httpcmd (http://ESP-IP/control?cmd=pcd8544cmd,blOn)
+//      - BackLight off via httpcmd (http://ESP-IP/control?cmd=pcd8544cmd,blOff)
+//      - Clear Display via httpcmd (http://ESP-IP/control?cmd=pcd8544cmd,clear)
+//      - Send Text via httpcmd     (http://ESP-IP/control?cmd=pcd8544,1,Hello World!)  // 1 : line#
+//        Send Text via http-request only works for the empty lines in ESPEasy-Webinterface!
+// - Pin connections:
+//       SPI interface:
+//       Example of tested configuration:
+//       RST  -      => LCD_pin_1 reset connected to Vcc with 10k resistor
+//       CE  GPIO-5  => LCD_pin_2 chip select
+//       DC  GPIO-32 => LCD_pin_3 Data/Command select
+//       DIN GPIO-23 => LCD_pin_4 Serial data 
+//       CLK GPIO-18 => LCD_pin_5 Serial clock out
+//       In hardware tab; 
+//       - enable SPI; 
+//       - Select VSPI:CLK=GPIO-18, MISO=GPIO-19, MOSI=GPIO-23 
+//         (miso not used)
+
+// ToDo:
+// - different digit-size within a line....
+// - Choice of usable fonts on web-webform.
+// - Remove webform-line "1st GPIO"
+//
+// Hardware note:
+// It's often seen that pins are connected via (10k) risistors.
+// Sometimes a screen will not work with these risistors. In that case, connect the display without resistors. 
+// That works well. However, I have no long-term experience with this.
+
+#define PLUGIN_208
+#define PLUGIN_ID_208 208
+#define PLUGIN_NAME_208 "Display - LCD PCD8544 (Nokia 5110) [Test]"
+#define PLUGIN_VALUENAME1_208 "Backlight"
+#define PLUGIN_VALUENAME2_208 "Contrast"
+#define PLUGIN_VALUENAME3_208 "Rotation"
+
+#include <Adafruit_GFX.h>
+#include <Adafruit_PCD8544.h>
+#define lcd_lines 6
+#define digits_per_template_line 48
+#define digits_per_display_line 14
+int8_t aa = 2;
+int8_t ab = 3;
+
+Adafruit_PCD8544 *lcd3 = nullptr;
+
+char html_input [lcd_lines][digits_per_display_line];
+
+boolean Plugin_208(byte function, struct EventStruct *event, String& string){
+boolean success = false;
+
+  switch (function)
+  {
+    case PLUGIN_DEVICE_ADD:{
+        Device[++deviceCount].Number = PLUGIN_ID_208 ;
+        Device[deviceCount].Type = DEVICE_TYPE_SINGLE;

```

Comment:
If you use the Device[n].Type `DEVICE_TYPE_SPI2` and use the `PLUGIN_GET_DEVICEGPIONAMES` case in your plugin, you get the PIN setup for 2 pins for 'free'. Examples are P095 and P096 (in the mega repository).

Reply:
Wow, thats fun. Source is changed in accordance with your advice.

Reply:
![Knipsel](https://user-images.githubusercontent.com/79747798/111059982-f1693a00-8499-11eb-8485-56e7cd680662.JPG)